feat: add C ndarray interface and refactor implementation for `stats/base/dvariancepn`

0PrashantYadav0 opened the pull request:
-

Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C ndarray interface for stats/base/dvariancepn
-   refactors JavaScript implementation
-   updates tests, docs, examples and benchmarks

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add C ndarray interface and refactor implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@Planeshifter @kgryte Please review this pr as my other pr are depended on this on #4687 and #4685. Thank you.

0PrashantYadav0 committed:
fix: updated test values

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 commented:
I have to make changes in test input value as well as the provide test doesn't match with other. 

@Planeshifter @kgryte Please review this pr. Thank you.

0PrashantYadav0 committed:
fix: updated test values

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: failed
---

0PrashantYadav0 committed:
fix: in test files

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 committed:
fix: updated input array value

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/src/main.c:
```diff
@@ -53,14 +69,14 @@ double stdlib_strided_dvariancepn( const int64_t N, const double correction, con
 	}
 	// Compute an estimate for the mean:
-	mu = stdlib_strided_dsumpw( N, X, stride ) / dN;
+	mu = stdlib_strided_dsumpw( N, X, strideX ) / dN;
```
```suggestion
	mu = API_SUFFIX(stdlib_strided_dsumpw_ndarray)( N, X, strideX, offsetX ) / dN;
```

aayush0325 replied:
any function call that has `CBLAS_INT` as one of it's parameters should be wrapped in `API_SUFFIX` everywhere

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/src/main.c:
```diff
@@ -53,14 +69,14 @@ double stdlib_strided_dvariancepn( const int64_t N, const double correction, con
 
-	if ( stride < 0 ) {
-		ix = (1-N) * stride;
+	if ( strideX < 0 ) {
+		ix = (1-N) * strideX;
 	} else {
 		ix = 0;
 	}
```
```suggestion
ix = offsetX;
```

aayush0325 replied:
ndarray interfaces should support custom offsets

aayush0325 reviewed (changes requested):
pointed out some inaccuracies in the C implementation of the ndarray interface in `src/main.c`, it's worth working on this and passing the CI before further review

0PrashantYadav0 committed:
fix: updated function to support ndarray

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

aayush0325 reviewed (changes requested):
I would prefer that you avoid making changes unless necessary while refactoring. Kindly minimize all unnecessary diff

aayush0325 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/test/test.ndarray.native.js:
```diff
@@ -188,12 +183,11 @@ tape( 'if provided a `stride` parameter equal to `0`, the function returns `0`',
 
 	x = new Float64Array([
-		2.0,
+		1.0,
```
I don't see any need to change this

0PrashantYadav0 commented:
Thank you so much @aayush0325 for help. @Planeshifter @kgryte Please review this pr and merge this coz other my other 2 pr are depended on this one.

0PrashantYadav0 committed:
fix: revert changes in test inputs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
````diff
@@ -224,6 +209,125 @@ console.log( v );
+```c
+const double x[] = { 1.0, -2.0, 2.0 };
+
+double v = stdlib_strided_dvariancepn_ndarray( 3, 1.0, x, 1, 0 );
````
```suggestion
double v = stdlib_strided_dvariancepn_ndarray( 4, 1.0, x, 1, 0 );
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
````diff
@@ -224,6 +209,125 @@ console.log( v );
+Computes the [variance][variance] of a double-precision floating-point strided array using a two-pass algorithm and alternative indexing semantics.
+
+```c
+const double x[] = { 1.0, -2.0, 2.0 };
````
```suggestion
const double x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 }
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
```diff
@@ -224,6 +209,125 @@ console.log( v );
+const double x[] = { 1.0, -2.0, 2.0 };
+
+double v = stdlib_strided_dvariancepn_ndarray( 3, 1.0, x, 1, 0 );
+// returns ~4.3333
```
@0PrashantYadav0 You might want to change the return value according to the new array and Value of N.
Since these values come directly from `example.c`

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
````diff
@@ -224,6 +209,125 @@ console.log( v );
+Computes the [variance][variance] of a double-precision floating-point strided array using a two-pass algorithm.
+
+```c
+const double x[] = { 1.0, -2.0, 2.0 };
````
similarly for this as well.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/docs/repl.txt:
```diff
@@ -31,8 +31,8 @@
-    stride: integer
-        Index increment.
+    strideX: integer
+        Stride Length.
```
```suggestion
        Stride length.
```

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/docs/repl.txt:
```diff
@@ -89,10 +86,10 @@
-    stride: integer
-        Index increment.
+    strideX: integer
+        Stride Length.
```
```suggestion
        Stride length.
```

0PrashantYadav0 committed:
chore: updated values as per given review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@Neerajpathak07 Thank you for your review. I make the changes you asked for.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
```diff
@@ -224,6 +209,125 @@ console.log( v );
+const double x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 }
+
+double v = stdlib_strided_dvariancepn( 4, 1.0, x, 1, 0 );
+// returns ~6.666667
```
@kgryte 
just a quick question I did the math for this comes out to be 1.25 is this accurate??

0PrashantYadav0 replied:
Yes i got this output after running example.c file with the same input.

Neerajpathak07 replied:
@Planeshifter any views on this!

kgryte replied:
@0PrashantYadav0 I am not sure how you are getting your answers. From the REPL,

```
In [1]: var x = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );

In [2]: base.strided.dvariancepn( x.length, 1.0, x, 1 )
Out[2]: 6

In [3]: base.strided.dvariancepn( 4, 1.0, x, 1 )
Out[3]: 1.6666666666666667

In [4]: base.strided.dvariancepn( 4, 0.0, x, 1 )
Out[4]: 1.25

In [5]: base.strided.dvariancepn( 8, 0.0, x, 1 )
Out[5]: 5.25
```

None of the answers match what you are displaying. Either match the JS examples in the README above OR do the more intuitive thing of taking the variance over the entire array when the stride is `1`. Regardless, your displayed answers need to be fixed.

Neerajpathak07 commented:
@0PrashantYadav0 
Would suggest you to make similar changes in your other PR's related to ndarray interface. 
As it makes it difficult for the maintainers to review it, had similar issues in my PR so thought of advising you to prevent some thing similar

0PrashantYadav0 commented:
@Neerajpathak07 Sure i will make changes in all my PR's related to ndarray interface.

0PrashantYadav0 commented:
@Planeshifter @kgryte Please review this as i make all the changes requested.

Neerajpathak07 commented:
@0PrashantYadav0 LGTM

0PrashantYadav0 commented:
@kgryte @Planeshifter Please review and merge this pr.

kgryte commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
````diff
@@ -224,6 +209,125 @@ console.log( v );
+```c
+const double x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 }
+
+double v = stdlib_strided_dvariancepn( 4, 1.0, x, 1, 0 );
````
```suggestion
double v = stdlib_strided_dvariancepn( 8, 1.0, x, 1 );
```

0PrashantYadav0 committed:
chore: updated inputs in docs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@kgryte Make the changes you asked for please review. Thank you.

0PrashantYadav0 commented:
@kgryte Please review this pr. Thank you.

0PrashantYadav0 commented:
@kgryte Will you review this pr please. Thank you.

Neerajpathak07 commented:
@0PrashantYadav0 apparently the maintainers are busy with the GSoC org registrations which should be done by the end of the week. Please wait for the time being they'll surely take a look at it.

0PrashantYadav0 commented:
No worries. Thank you for informing me. @Neerajpathak07.

aayush0325 reviewed (approved):
lgtm

kgryte commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
```diff
@@ -224,6 +209,125 @@ console.log( v );
+const double x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 }
+
+double v = stdlib_strided_dvariancepn( 8, 1.0, x, 1 );
+// returns 6
```
```suggestion
// returns 6.0
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
````diff
@@ -224,6 +209,125 @@ console.log( v );
+```c
+const double x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 }
+
+double v = stdlib_strided_dvariancepn_ndarray( 4, 1.0, x, 1, 0 );
````
```suggestion
double v = stdlib_strided_dvariancepn_ndarray( 4, 1.0, x, 2, 0 );
```

Please be sure to double-check your examples.

kgryte commented on lib/node_modules/@stdlib/stats/base/dvariancepn/README.md:
```diff
@@ -224,6 +209,125 @@ console.log( v );
+    const int strideX = 2;
+
+    // Compute the variance:
+    double v = stdlib_strided_dvariancepn( N, 1, x, strideX );
```
```suggestion
    double v = stdlib_strided_dvariancepn( N, 1.0, x, strideX );
```

Be consistent in your use of decimals.

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dvariancepn/src/main.c:
```diff
@@ -53,25 +69,21 @@ double stdlib_strided_dvariancepn( const int64_t N, const double correction, con
-	} else {
-		ix = 0;
-	}
+	ix = offsetX;
```
```suggestion
	ix = offsetX;

```

kgryte committed:
style: insert empty line before comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
